lib/vnet/app_resolver.go

```diff
@@ -394,13 +395,34 @@ func (h *tcpAppHandler) getOrInitializeLocalProxy(ctx context.Context, localPort
 // HandleTCPConnector handles an incoming TCP connection from VNet by passing it to the local alpn proxy,
 // which is set up with middleware to automatically handler certificate renewal and re-logins.
 func (h *tcpAppHandler) HandleTCPConnector(ctx context.Context, localPort uint16, connector func() (net.Conn, error)) error {
+	if len(h.app.GetTCPPorts()) > 0 {
+		tcpPorts := types.PortRanges(h.app.GetTCPPorts())
```

Comment:
is it too late to make app.GetTCPPorts() return a types.PortRanges?

Reply:
It wasn't included in any public release of Teleport, so I think it should be fine.